lib/qesapdeployment.pm

```diff
@@ -745,4 +749,135 @@ sub qesap_get_az_resource_group {
     return $result;
 }
 
+=head3 qesap_get_az_address_range
+
+Query the existing maintenance deployments for azure
+and determine a name with a free adress range number
+for the current deployment
+
+=cut
+
+sub qesap_get_az_address_range {
+    my $worker = get_var("WORKER_INSTANCE");
+
+    my $offset = ($worker - 1) * 8;
+
+    my $ip2 = int($offset / 256);
+    my $ip3 = $offset % 256;
+
+    my $vnet_address_range = sprintf("10.%d.%d.0/21", $ip2, $ip3);
+    my $subnet_address_range = sprintf("10.%d.%d.0/24", $ip2, $ip3);
+
+    return (
+        vnet_address_range => $vnet_address_range,
+        subnet_address_range => $subnet_address_range,
+    );
+}
+
+=head3 qesap_az_vnet_peering
+
+    Performs peering between the cluster and IBS mirror on Azure.
+
+=over 3
+
+=item B<SOURCE_GROUP> - resource group of source
+
+=item B<SOURCE_VNET> - vnet of source
+
+=item B<TARGET_GROUP> - resource group of target
+
+=item B<TARGET_VNET> - vnet of target
+
+=item B<VERBOSE> - wether output will be verbose or not
+
+=back
+=cut
+
+sub qesap_az_vnet_peering {
+    my (%args) = @_;
+    croak 'Missing mandatory source_group argument' unless $args{source_group};
+    croak 'Missing mandatory source_vnet argument' unless $args{source_vnet};
```

Comment:
can we think about an `az` query to obtain `vnet` from the `resource_group` ? So reduce the input arguments from 4 to 2?
```
az network vnet list -g ${MY_AZ_GROUP} --query "[0].name" -o tsv
```

Something like this (take care that I'm not sure about the zero in `[0].name`

Reply:
We probably can - but, this will be replaced by a script soon, right? 

So we could probably do that in the script, to avoid the double effort. Better sooner than later (maybe even expand the scope of this ticket to add the script as well?)